src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -149,22 +176,30 @@ protected void trySearchNewRecipe() {
         Recipe currentRecipe = null;
         IItemHandlerModifiable importInventory = getInputInventory();
         IMultipleTankHandler importFluids = getInputTank();
-        if (previousRecipe != null && previousRecipe.matches(false, importInventory, importFluids)) {
-            //if previous recipe still matches inputs, try to use it
-            currentRecipe = previousRecipe;
-        } else {
-            boolean dirty = checkRecipeInputsDirty(importInventory, importFluids);
-            if (dirty || forceRecipeRecheck) {
-                this.forceRecipeRecheck = false;
-                //else, try searching new recipe for given inputs
-                currentRecipe = findRecipe(maxVoltage, importInventory, importFluids);
-                if (currentRecipe != null) {
-                    this.previousRecipe = currentRecipe;
-                }
-            }
+
+        // see if the last recipe we used still works
+        if (this.previousRecipe != null && this.previousRecipe.matches(false, importInventory, importFluids))
+            currentRecipe = this.previousRecipe;
+        // If there is no active recipe, then we need to find one.
+        else {
+            metaTileEntity.setInputsDirty(false);
```

Comment:
Is setting the inputs as not dirty here correct? We will be finding a new recipe, thus causing the inputs to change, meaning that unless the output is full, or the recipe lookup fails, the inputs will always be dirty.